File: agents-backup/gatekeeper/test_hover.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch();
  const page = await browser.newPage({ viewport: { width: 1280, height: 900 } });
  await page.goto('https://hwzhxs.github.io/squad-landing/', { waitUntil: 'networkidle' });
  
  const thinker = page.locator('text=Thinker').first();
  await thinker.scrollIntoViewIfNeeded();
  await page.waitForTimeout(500);
  
  const bbox = await thinker.boundingBox();
  console.log('Thinker bbox:', JSON.stringify(bbox));
  
  await page.screenshot({ path: 'before_hover.png' });
  
  // Hover over the card
  await page.mouse.move(bbox.x + bbox.width / 2, bbox.y);
  await page.waitForTimeout(3000);
  
  await page.screenshot({ path: 'after_hover.png' });
  
  // Check videos
  const videos = await page.$$('video');
  console.log(`Video elements: ${videos.length}`);
  for (let i = 0; i < videos.length; i++) {
    const info = await videos[i].evaluate(el => ({
      src: el.src || '',
      sourceSrc: el.querySelector('source')?.src || '',
      visible: el.offsetParent !== null,
      paused: el.paused,
      opacity: getComputedStyle(el).opacity,
      display: getComputedStyle(el).display,
      currentTime: el.currentTime,
      readyState: el.readyState,
      width: el.videoWidth,
      height: el.videoHeight
    }));
    console.log(`Video ${i}:`, JSON.stringify(info));
  }
  
  await browser.close();
})();

File: agents/gatekeeper/test_hover.py
from playwright.sync_api import sync_playwright
import time

with sync_playwright() as p:
    browser = p.chromium.launch()
    page = browser.new_page(viewport={'width': 1280, 'height': 900})
    page.goto('https://hwzhxs.github.io/squad-landing/', wait_until='networkidle')
    
    thinker = page.locator('text=Thinker').first
    thinker.scroll_into_view_if_needed()
    time.sleep(0.5)
    
    bbox = thinker.bounding_box()
    print(f'Thinker text bbox: {bbox}')
    
    # Screenshot before hover
    page.screenshot(path='before_hover.png')
    
    # Hover over the card area
    page.mouse.move(bbox['x'] + bbox['width']/2, bbox['y'])
    time.sleep(3)
    
    page.screenshot(path='after_hover.png')
    
    # Check for video elements
    videos = page.query_selector_all('video')
    print(f'Video elements found: {len(videos)}')
    for i, v in enumerate(videos):
        src = v.get_attribute('src') or ''
        source = v.query_selector('source')
        source_src = source.get_attribute('src') if source else ''
        visible = v.is_visible()
        paused = v.evaluate('el => el.paused')
        opacity = v.evaluate('el => getComputedStyle(el).opacity')
        display = v.evaluate('el => getComputedStyle(el).display')
        print(f'Video {i}: src={src}, source={source_src}, visible={visible}, paused={paused}, opacity={opacity}, display={display}')
    
    browser.close()
